streamalert/scheduled_queries/config/services.py

```diff
@@ -20,137 +20,182 @@
+    def state_manager(self):
+        return self._service_container.get('state_manager')
+
+    @property
+    def clock(self):
+        return self._service_container.get('clock')
+
+    def create_step_function_state_manager(self):
+        return StepFunctionStateManager(
+            self.state_manager,
+            self.logger,
+            self.clock
+        )
+
+
 # pylint: disable=too-many-statements
 def configure_container(container):
     """Configures the container
 
     Params:
         container (ServiceContainer)
     """
-
-    def make_command_processor(_container):
-        return CommandProcessor(
-            logger=_container.get('logger'),
-            kinesis=_container.get('streamalert_forwarder'),
-            state_manager=_container.get('state_manager'),
-            manager_factory=_container.get('query_pack_manager_factory')
-        )
-
-    def make_logger(_container):
-        logger = logging.getLogger(_container.get_parameter('command_name'))
-        logger.setLevel(_container.get_parameter('log_level').upper())
-        logging.basicConfig(
-            format='%(name)s [%(levelname)s]: [%(module)s.%(funcName)s] %(message)s'
-        )
-        return logger
-
-    def make_kinesis(_container):
-        return KinesisClient(
-            logger=_container.get('logger'),
-            client=_container.get('boto3_kinesis_client'),
-            kinesis_stream=_container.get_parameter('kinesis_stream')
-        )
-
-    def make_cache(_container):
-        cache = StateManager(
-            logger=_container.get('logger')
-        )
-
-        return cache
-
-    def make_athena(_container):
-        return AthenaClient(
-            logger=_container.get('logger'),
-            client=_container.get('boto3_athena_client'),
-            database=_container.get_parameter('athena_database'),
-            results_bucket=_container.get_parameter('athena_results_bucket')
+    container.register(ServiceDefinition('command_processor', _make_command_processor))
+    container.register(ServiceDefinition('logger', _make_logger))
+    container.register(ServiceDefinition('streamalert_forwarder', _make_kinesis))
+    container.register(ServiceDefinition('state_manager', _make_cache))
+    container.register(ServiceDefinition('athena', _make_athena))
+    container.register(ServiceDefinition('query_parameter_generator', _make_param_generator))
+    container.register(ServiceDefinition('query_pack_repository', _make_query_pack_repo))
+    container.register(ServiceDefinition('query_pack_manager_factory', _make_query_pack_factory))
+    container.register(ServiceDefinition('query_pack_execution_context', _make_execution_context))
+    container.register(ServiceDefinition('clock', _make_clock))
+    container.register(ServiceDefinition('boto3_athena_client', _make_boto3_athena_client))
+    container.register(ServiceDefinition('boto3_kinesis_client', _make_boto3_kinesis_client))
+
+
+def _make_command_processor(_container):
```

Comment:
is there a reason for the underscore prefix on these arg names? I'm unsure where this convention comes from or signifies?

Reply:
No reason anymore. They used to be nested methods but I pulled them out to global namespace and didn't remember to remove the underscores

Reply:
Fixed https://github.com/airbnb/streamalert/pull/1130/commits/db7d0f1cb5162140f42d9dd28b700739d0e1f107